samza-core/src/main/java/org/apache/samza/config/JobConfig.java

```diff
@@ -128,6 +128,9 @@
   public static final String COORDINATOR_STREAM_FACTORY = "job.coordinatorstream.config.factory";
   public static final String DEFAULT_COORDINATOR_STREAM_CONFIG_FACTORY = "org.apache.samza.util.DefaultCoordinatorStreamConfigFactory";
 
+  // enables application master isolation
+  public static final String SAMZA_APPLICATION_MASTER_ISOLATION_ENABLED = "samza.application.master.isolation.enabled";
```

Comment:
Maybe we should include the word `dependency` in the config name while referring to isolation ?
Also do you think we should maybe rename this config to be more generic like
`samza.cluster.based.job.coordinator.dependency.isolation.enabled` ?
If not, maybe move it to `YarnConfig.java` since this is YARN specific ?

Reply:
Good point. I'll use the "cluster based job coordinator dependency" terminology.